src/Personal/actionCreator.js

```diff
@@ -0,0 +1,79 @@
+import { BACKEND_URL } from 'react-native-dotenv';
+
+export const actionType = {
+  BACKUP_START: 'BACKUP_START',
+  BACKUP_DATA: 'BACKUP_DATA',
+  BACKUP_FAILED: 'BACKUP_FAILED',
+  BACKUP_INITIAL: 'BACKUP_INITIAL',
+  SIGNUP_START: 'SIGNUP_START',
+  SIGNUP_SUCCESSFUL: 'SIGNUP_SUCCESSFUL',
+  SIGNUP_FAILED: 'SIGNUP_FAILED',
+};
+
+export function backupStart() {
+  return {
+    type: actionType.BACKUP_START,
+  };
+}
+
+export function backupData() {
+  return {
+    type: actionType.BACKUP_DATA,
+  };
+}
+
+export function backupFailed() {
+  return {
+    type: actionType.BACKUP_FAILED,
+  };
+}
+
+export function initialBackup() {
+  return {
+    type: actionType.BACKUP_INITIAL,
+  };
+}
+
+export function signupStart() {
+  return {
+    type: actionType.SIGNUP_START,
+  };
+}
+
+export function signupSuccessful(userInfo) {
+  return {
+    type: actionType.signupSuccessful,
+    payload: userInfo,
+  };
+}
+
+export function signupFailed() {
+  return {
+    type: actionType.signupFailed,
+  };
+}
+
+export const requestBackup = (transactions) => (dispatch) => {
```

Comment:
You can get access to the store in the dispatch mathods so you dont need the transactions as parameters. 
```js
export const requestBackup = (transactions) => (dispatch, getState()) => {
  const { transactions } = getState();
  ,,,
```

Reply:
Just tried, but it shows the error `assigning to rvalue`

Reply:
sorry, my bad:
```js
export const requestBackup = (transactions) => (dispatch, getState) => {
  const { transactions } = getState();
  ...
```

Of course we dont need to call the method in the arguments list

Reply:
Thank you, have fixed